src/Plugins/RpcServer/ParameterConverter.cs

```diff
@@ -219,23 +259,22 @@ internal static SignersAndWitnesses ToSignersAndWitnesses(this JToken json, byte
         /// <param name="addressVersion">The address version to use for the signers.</param>
         /// <returns>A Signer array.</returns>
         /// <exception cref="RpcException">Thrown when the JSON array is invalid or max allowed witness exceeded.</exception>
-        private static Signer[] ToSigners(this JArray json, byte addressVersion)
+        internal static Signer[] ToSigners(this JArray json, byte addressVersion)
         {
             if (json.Count > Transaction.MaxTransactionAttributes)
                 throw new RpcException(RpcError.InvalidParams.WithData("Max allowed signers exceeded."));
 
-            var ret = json.Select(u => new Signer
+            var signers = new Signer[json.Count];
+            for (var i = 0; i < json.Count; i++)
             {
-                Account = u["account"].AsString().AddressToScriptHash(addressVersion),
-                Scopes = (WitnessScope)Enum.Parse(typeof(WitnessScope), u["scopes"]?.AsString()),
-                AllowedContracts = ((JArray)u["allowedcontracts"])?.Select(p => UInt160.Parse(p.AsString())).ToArray() ?? [],
-                AllowedGroups = ((JArray)u["allowedgroups"])?.Select(p => ECPoint.Parse(p.AsString(), ECCurve.Secp256r1)).ToArray() ?? [],
-                Rules = ((JArray)u["rules"])?.Select(r => WitnessRule.FromJson((JObject)r)).ToArray() ?? [],
-            }).ToArray();
+                if (json[i] is null || json[i] is not JObject obj)
+                    throw new RpcException(RpcError.InvalidParams.WithData($"Invalid Signer at {i}."));
+                signers[i] = ToSigner(obj, addressVersion);
```

Comment:
Please don't write it like this. This isn't our style.
```suggestion
                signers[i] = obj.ToSigner(addressVersion);
```